Add note on PHP 8.1

joa-pan opened the pull request:
<!--
Pull requests should be opened in a branch off the `main` branch.

For more information on contributing to Pantheon documentation:
- [Contributor Guidelines](https://pantheon.io/docs/contribute)
- [Style Guide](https://pantheon.io/docs/style-guide)
- and the [Google developer documentation style guide](https://developers.google.com/style) for formatting recommendations when contributing to the docs.

**Note:** Please fill out the PR template to ensure proper processing and release timing. If you're not sure about a section, leave it empty.
-->

Closes #7023 

## Summary

**[PHP versions](https://pantheon.io/docs/php-versions)** - Add compatibility info for PHP 8.1

## Effect

<!-- Use this section to detail the changes summarized above, or remove if not needed -->

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

<!-- Remove if not needed -->
The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

<!-- If this PR relies on other work before it should be merged or if it should be merged after a certain date, detail that here. -->

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [ ] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

pantheondocs commented:
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-7234](https://pr-7234--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `46` | `46` | `0` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `98` | `98` | `0` |

Lighthouse Audit Reports:

- [`pr-7234` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/35581/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/35581/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-7234--pantheon-docs.my.pantheonfrontend.website/docs)

CdrMarks commented:
@joa-pan I'd make the following changes:

1. Change the Alert Title to be "Compatibility Notes"
2. Change the Popover content to "See the Compatibility Notes section below".
3. Add the Popover to PHP version 8.0 in addition to 8.1
5. In the Alert body, separate the WordPress compatibility note out into its own line.  Add a link to https://make.wordpress.org/core/2022/01/10/wordpress-5-9-and-php-8-0-8-1/ and include, in your own words, "
Work is still ongoing and any open tickets in the post you linked are still relevant."

joa-pan commented:
Thanks @CdrMarks. I updated the info based on your recommendations. I think we should stick with "Compatibility Note" as the admonition is  "Note" (just to be consistent), but I incorporated your other suggestions . Thanks a million for the review

CdrMarks commented:
[Preview](https://pr-7234--pantheon-docs.my.pantheonfrontend.website/docs/php-versions)

CdrMarks commented:
@joa-pan I think it's redundant to have any details in the popover.  Can you direct the reader to the Compatibility Considerations section.  

I'd recommend you use an Alert box so that the Compatibility Considerations stands out.

Can you also link 8.1 in the table to https://v81-php-info.pantheonsite.io/ ?

whitneymeredith commented:
LGTM

CdrMarks commented on source/content/php-versions.md:
```diff
@@ -55,6 +55,15 @@ End-of-life (**EOL**) versions are available on the platform but no longer under
 
+<Alert title="Note" type="info">
+
+PHP 8.1 contains new features, performance improvements, deprecations, and backward compatibility breaks and was released November 25, 2021. 
```
I like this part and would have left it in place.

CdrMarks commented on source/content/php-versions.md:
```diff
@@ -28,8 +28,8 @@ Changes made to the `pantheon.yml` file on a branch **are not** detected when cr
 | --------------------------------------------:|:-----------:| ------- |
-| 8.1 | ➖ | Available <Popover title="New Relic" content="New Relic is not supported in PHP 8.1." /> |
-| [8.0](https://v80-php-info.pantheonsite.io/) | ✅          | Active  |
+| [8.1](https://v81-php-info.pantheonsite.io/)| ➖ | Available <Popover title="Compatibility Note" content="WordPress is not fully compatible with PHP 8.1. New Relic is not supported in PHP 8.1." /> |
+| [8.0](https://v80-php-info.pantheonsite.io/) | ✅          | Active <Popover title="Compatibility Note" content="WordPress is not fully compatible with PHP 8.0." /> |
```
I believe the content here should direct the reader to the Compatibility Considerations section so we're not repeating ourselves.

joa-pan replied:
Thank you @CdrMarks. It's a good idea to direct users to that section. I'll open a new PR to add that change. Thanks a miilion

CdrMarks reviewed (commented):
@joa-pan @whitneymeredith 
I see this was merged, but feel there is still work to be done here.
